### PR TITLE
Don't include GraalVM dependencies in lib directory

### PR DESCRIPTION
**Pull request by zakkak**
In preparation for oracle/graal#5232

Since GraalVM dependencies are provided by GraalVM itself there is no need to include them in the lib directory of thin jars or in uber jars.

Supersedes https://github.com/quarkusio/quarkus/pull/28647

CI run: https://github.com/graalvm/mandrel/actions/runs/3269063301

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java [hunk 1]
```diff
@@ -429,6 +430,9 @@ private static boolean includeAppDep(ResolvedDependency appDep, Optional<Set<Art
         if (!appDep.isJar()) {
             return false;
         }
+        if (appDep.getGroupId().equals("org.graalvm.nativeimage") || appDep.getGroupId().equals("org.graalvm.sdk")) {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:433]: Do we need to only do this for a specific GraalVM version or will this work for all versions we support?

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:433]: It should work for all versions.

* Tested with 23.0-dev in https://github.com/graalvm/mandrel/actions/runs/3269063301
* And being tested with 22.2 in this PR.
* 22.3 should be no different.

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:433]: Thanks!

**Review by geoand — APPROVED** [review]

**Comment by zakkak** [comment]
I see a lot of timeout errors in the JVM tests. I don't see how these could relate to this change, but I am also hesitant to believe it's coincidence. Should we re-runt the tests?

**Comment by geoand** [comment]
Yeah, best re-run them

**Comment by geoand** [comment]
It seems like the test failures are genuine

**Comment by zakkak** [comment]
> It seems like the test failures are genuine

Yes, apparently there are some dependencies on graal-sdk in JVM mode. I altered the PR to only skip the dependencies in native builds.

New CI run with https://github.com/oracle/graal/pull/5232 : https://github.com/graalvm/mandrel/actions/runs/3273144262

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
@zakkak is it required for GraalVM 22.3 (and so should be backported) or is it 22.4+?

**Comment by zakkak** [comment]
No, this will only be required for GraalVM 23.0

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java [hunk 2]
```diff
@@ -973,6 +973,14 @@ private NativeImageSourceJarBuildItem buildNativeImageThinJar(CurateOutcomeBuild
 
             log.info("Building native image source jar: " + runnerJar);
 
+            // Remove svm and graal-sdk artifacts as they are provided by GraalVM itself
+            if (classLoadingConfig.removedArtifacts.isEmpty()) {
+                classLoadingConfig.removedArtifacts = Optional.of(new ArrayList<>(2));
+            }
+            List<String> removedArtifacts = classLoadingConfig.removedArtifacts.get();
+            removedArtifacts.add("org.graalvm.nativeimage:svm");
+            removedArtifacts.add("org.graalvm.sdk:graal-sdk");
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:982]: Can't say I'm excited about us actually updating the config. But I suppose it shouldn't be a huge problem in practice.

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java:982]: I went this way to reuse the existing code and avoid introducing more parameters and checks that would essentially do the same thing. As this is not urgent please let me know if you can think of any better alternatives.

**Comment by zakkak** [comment]
Test failures seem unrelated to the PR.

@gsmet we would need a fix sooner than later since https://github.com/oracle/graal/pull/5232 is now merged upstream and is causing CI failures (when testing with 23.0-dev), please advise on how to proceed.

**Comment by jerboaa** [comment]
> @gsmet we would need a fix sooner than later since [oracle/graal#5232](https://github.com/oracle/graal/pull/5232) is now merged upstream and is causing CI failures (when testing with 23.0-dev), please advise on how to proceed.

Yes, please. Our integration tests are all red because of this and are at risk of missing other breakage because of it.

**Comment by jerboaa** [comment]
Thank you!

**Pull request merged**